app/javascript/packages/form-steps/form-steps.tsx

```diff
@@ -169,6 +170,17 @@ function useStepTitle(step: FormStep, titleFormat?: string) {
   }, [step]);
 }
 
+function trackStepEvents(step: FormStep) {
+  useEffect(() => {
+    if (step && step.name === 'personal_key') {
```

Comment:
I was getting some errors when running tests about the step not existing so I added && step to these if statements

Reply:
> I was getting some errors when running tests about the step not existing so I added `&& step` to these if statements to get around that.

Ideally that seems like something our TypeScript types would be catching for us. Looking at it locally, I see it's with the ["empty steps array" test case](https://github.com/18F/identity-idp/blob/a013cede5d029ea4edb38a92223cca1ae19280c1/app/javascript/packages/form-steps/form-steps.spec.tsx#L125). TypeScript seems to be a bit too trusting that [the array access](https://github.com/18F/identity-idp/blob/a013cede5d029ea4edb38a92223cca1ae19280c1/app/javascript/packages/form-steps/form-steps.tsx#L235) would result in an object.

<img width="295" alt="image" src="https://user-images.githubusercontent.com/1779930/165515382-108cf5dc-7228-4f66-abc2-7b3fe0d20df0.png">

One thing we might want to do is explicitly set the type of that variable so that we can have TypeScript provide additional assurances that it might be undefined.

e.g.

```ts
const step = steps[stepIndex] as FormStep | undefined;
```